lms/models/application_instance.py

```diff
@@ -66,6 +87,41 @@ def get_by_consumer_key(cls, db, consumer_key):
         """Return the ApplicationInstance with the given consumer_key or None."""
         return db.query(cls).filter_by(consumer_key=consumer_key).one_or_none()
 
+    def update_lms_data(self, params: dict):
```

Comment:
Not a great interface but it's meant to be called from a couple of places and this is better that doing request.params['...'] for each one in both.

Arguably taking `tool_consumer_instance_guid` explicitly and then the rest could be a better one.